### PR TITLE
Add examples to documentation: Part-2 (c-d)

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
Part of https://github.com/r-lib/lintr/issues/1492

### CODE REVIEW

File: R/class_equals_linter.R [hunk 1]
```diff
@@ -5,7 +5,30 @@
...
+#' )
+#'
+#' # okay
+#' lint(
+#'   text = "is_lm <- inherits(x, 'lm')",
+#'   linters = class_equals_linter()
+#' )
+#'
+#' lint(
+#'   text = "if (inherits(x, 'character')) stop('no')",
```
Comment (MichaelChirico) [inline R/class_equals_linter.R:29]: I wouldn't use this, `inherits(x, "character")` is itself an antipattern... use `is.character()` instead?

File: R/commas_linter.R [hunk 2]
```diff
@@ -2,6 +2,34 @@
...
+#' )
+#'
+#' lint(
+#'   text = "mean(x,trim = 0.2,na.rm = TRUE)",
+#'   linters = commas_linter()
+#' )
+#'
+#' # okay
+#' lint(
+#'   text = "switch(op, x = foo, y = bar)",
```
Comment (MichaelChirico) [inline R/commas_linter.R:19]: since we're using `switch()` already, please add a case with an empty switch argument to show how it's done (`switch(op, x = , y = foo)`)

File: R/commas_linter.R [hunk 3]
```diff
@@ -2,6 +2,34 @@
...
+#'   linters = commas_linter()
+#' )
+#'
+#' lint(
+#'   text = "mean(x, trim = 0.2, na.rm = TRUE)",
+#'   linters = commas_linter()
+#' )
+#'
+#' lint(
+#'   text = "a[1, , 2, , 3]",
```
Comment (MichaelChirico) [inline R/commas_linter.R:29]: `[]` is a funny one with this linter, would help to add a "bad" example as well, e.g. `x[,, drop=TRUE]`

File: R/condition_message_linter.R [hunk 4]
```diff
@@ -1,9 +1,36 @@
 #' Block usage of `paste()` and `paste0()` with messaging functions using `...`
 #'
-#' `stop(paste0(...))` is strictly redundant -- `stop(...)` is equivalent.
-#'   `stop(...)` is also preferable to `stop(paste(...))`. The same applies to
-#'   all default condition functions, i.e., [stop()], [warning()], [message()],
-#'   and [packageStartupMessage()].
+#' Both `stop(paste0(...))` and `stop(paste(...))` are strictly redundant -- `stop(...)` is
```
Comment (MichaelChirico) [inline R/condition_message_linter.R:3]: Don't like the rewording... the point is that `stop(paste0(...))` is _exactly_ equivalent to `stop(...)`, but it's not quite so for `stop(paste(...))` because we have to add extra whitespace to account for `sep = " "`, so "strictly redundant" is too strong.

Maybe reword with that in mind?

File: R/condition_message_linter.R [hunk 5]
```diff
@@ -1,9 +1,36 @@
...
+#' )
+#'
+#' lint(
+#'   text = "warning(paste0('a string', ' another'))",
+#'   linters = condition_message_linter()
+#' )
+#'
+#' # okay
+#' lint(
+#'   text = "stop('a string', 'another')",
```
Comment (MichaelChirico) [inline R/condition_message_linter.R:21]: add the extra space here as mentioned so that the bad/good calls are more correspondent

File: R/cyclocomp_linter.R [hunk 6]
```diff
@@ -4,6 +4,20 @@
 #'
 #' @param complexity_limit Maximum cyclomatic complexity, default 15. Expressions more complex
 #' than this are linted. See [cyclocomp::cyclocomp()].
+#'
+#' @examples
+#' # will produce lints
+#' lint(
+#'   text = "if(TRUE) 1 else 2",
```
Comment (MichaelChirico) [inline R/cyclocomp_linter.R:11]: nit: isn't the spacing rule `if (TRUE) 1 else 2`?

File: tests/testthat/test-condition_message_linter.R [hunk 7]
```diff
@@ -38,6 +44,25 @@ test_that("condition_message_linter blocks simple disallowed usages", {
...
   )
 
+  # `sep` argument allowed, but only if it is different from default
+  expect_lint(
+    "stop(paste(x, sep = ' '))",
+    rex::rex("Don't use paste to build stop strings."),
+    condition_message_linter()
+  )
+
+  expect_lint(
```
Comment (MichaelChirico) [inline tests/testthat/test-condition_message_linter.R:54]: i don't think we need three tests for this, unless it's being done with a `patrick` parameterized test

File: R/condition_message_linter.R (hunk 4 above)
Comment (IndrajeetPatil) [reply R/condition_message_linter.R]: I have reworded this, but not sure if I have stuck the landing with wording. Happy to give it another try.

File: R/condition_message_linter.R [hunk 8]
```diff
@@ -1,9 +1,39 @@
 #' Block usage of `paste()` and `paste0()` with messaging functions using `...`
 #'
-#' `stop(paste0(...))` is strictly redundant -- `stop(...)` is equivalent.
-#'   `stop(...)` is also preferable to `stop(paste(...))`. The same applies to
-#'   all default condition functions, i.e., [stop()], [warning()], [message()],
-#'   and [packageStartupMessage()].
+#' This is because
```
Comment (MichaelChirico) [inline R/condition_message_linter.R:3]: I'd avoid a pronoun (`This`) to start the documentation, instead spell out what `This` is (note that in the actual Rd, this sentence is not connected to the title):

> This linter discourages combining condition functions like [stop()] with string concatenation functions [paste()] and [paste0()]. This is because.

Then below we don't need to link [stop()]

**Review by MichaelChirico — DISMISSED** [review]

File: tests/testthat/test-condition_message_linter.R (hunk 7 above)
Comment (IndrajeetPatil) [reply tests/testthat/test-condition_message_linter.R]: I have added parametrized tests now.

**Review by MichaelChirico — DISMISSED** [review]

**Review by MichaelChirico — APPROVED** [review]
